modules/apps/repository/repository-test/src/testIntegration/java/com/liferay/repository/test/RepositoryFactoryTest.java

```diff
@@ -31,29 +55,80 @@ public class RepositoryFactoryTest {
 	@ClassRule
 	@Rule
 	public static final AggregateTestRule aggregateTestRule =
-		new LiferayIntegrationTestRule();
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
 
 	@Before
 	public void setUp() throws Exception {
 		_group = GroupTestUtil.addGroup();
 	}
 
+	@Test
+	public void testAddFolderInRepositoryWhenSameRepositoryIdExistsInDifferentPartition()
+		throws Exception {
+
+		if (!DBPartition.isPartitionEnabled()) {
+			return;
+		}
+
+		Company company = CompanyTestUtil.addCompany();
+		com.liferay.portal.kernel.model.Repository repository1;
+
+		long companyId = CompanyThreadLocal.getCompanyId();
+		String userId = PrincipalThreadLocal.getName();
```

Comment:
This variable grouping looks odd. Can you group the variables that are used company/principal context apart from the actual test code? Also, make sure variable declaration order matches usage (e.g. you're declaring `repository1` before other variables it depends on (company, group, etc.).
```
Company company = CompanyTestUtil.addCompany();

User user = UserTestUtil.addUser(company);

long oldCompanyId = ...; // You're calling this `companyId`, but it is the old one, not the one for the company you just created.

String oldUserId = PrincipalThreadLocal.getName(); // The line break before this is important.

CompanyThreadLocal.setCompanyId(company.getCompanyId());

PrincipalThreadLocal.setName(user.getUserId());

try {
     ....
} finally {
     ....
}
```